user/test/com/google/gwt/emultest/java10/util/stream/CollectorsTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.gwt.emultest.java10.util.stream;
+
+import static com.google.gwt.emultest.java8.util.stream.CollectorsTest.applyItems;
+import static java.util.stream.Collectors.toUnmodifiableList;
+import static java.util.stream.Collectors.toUnmodifiableMap;
+import static java.util.stream.Collectors.toUnmodifiableSet;
+
+import com.google.gwt.emultest.java.util.EmulTestBase;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ * Tests for java.util.stream.Collectors Java 10 API emulation.
+ */
+public class CollectorsTest extends EmulTestBase {
+  private static <T> boolean unmodifiableCollection(Collection<T> c, T existingSample, T newSample) {
```

Comment:
It misses a lot of cases. Maybe we should refactor a bit in a follow up commit that introduces a helper class to check immutability of collections/sets, lists, maps. We now have many tests that need it: `CollectionsTest`, `ListTest`, `SetTest`, `MapTest`, `CollectorsTest`. Each doing it a bit differently and not covering everything.

Reply:
Good call, I'll file a followup. At least for the `List`-verification method, the simplest answer might be to make `CollectionsTest.doTestModificationsToList` public and static, and lean on it from here.

https://github.com/gwtproject/gwt/issues/9882

Reply:
>  at some point we have to assume that the levels below us are roughly sane

Yes, I had the thought that it might be good to just make the emulated immutable collection classes public and then simply do an instanceof check. If Collections.unmodifiableList/List.of and friends all use the same implementations, that would be fine I guess. Then we could make one test for the immutable implementations and all other tests check if that implementation has been used.